app/services/catalog/create_approval_request.rb

```diff
@@ -0,0 +1,66 @@
+require 'approval_api_client'
+
+module Catalog
+  class CreateApprovalRequest
+    attr_reader :order
+
+    def initialize(id)
+      @order = Order.find_by!(:id => id)
+
+      ApprovalApiClient.configure do |config|
+        config.api_key['x-rh-auth-identity'] = ManageIQ::API::Common::Request.current.request_id
+      end
+
+      # Set up the instance class-wide since there are N portfolio_items and workflows to request
+      @api_instance = ApprovalApiClient::RequestApi.new
+    end
+
+    def process
+      @order.order_items.each do |item|
+        approvals = send_approval_requests(item)
+
+        approvals.compact.each do |req|
+          @order.approval_requests << create_approval_request(req)
+        end
+      end
+
+      @order.update(:status => "Ordered", :ordered_at => Time.now.utc)
+
+      self
+    rescue ApprovalAPIClient::ApiError => e
+      Rails.logger.error("Error putting in approval Request for #{order.id}: #{e.message}")
+      raise
+    end
+
+    private
+
+    def send_approval_requests(item)
+      [].tap do |requests|
+        workflows(item.portfolio_item).each do |workflow|
+          requests << @api_instance.create_request(workflow, request_body_from(item))
+        end
+      end
+    end
+
+    def request_body_from(item)
+      ApprovalApiClient::RequestIn.new.tap do |request|
+        request.requester = ManageIQ::API::Common::Request.current.user.username
+        request.name      = item.portfolio_item.name
+        request.content   = item.service_parameters.to_json
```

Comment:
@lindgrenj6 These parameters have to be sanitized we can't send passwords and tokens to Approval Request and it might send it out in an email.

This PR https://github.com/ManageIQ/catalog-api/pull/46 already does most of the work that you trying to implement here

Reply:
@lindgrenj6 following up on @mkanoor's comment.  The filter parameters native to Rails should help with that out of the box.  https://api.rubyonrails.org/classes/ActionDispatch/Http/FilterParameters.html

Reply:
These parameters are based on the OpenServiceBroker API and they have datatypes that indicate that the type is encrypted field. But not all plans are adhering to it so we have to be do it based on data type and name.